### PR TITLE
mbs shared data

### LINKED ISSUE #33: mbox crd
As an User, I want one mbox CR to handle all mbox CRDs for me.

## Acceptance Criteria

* manage root CA secret - create one if it is not present
* manage shared koji PVC for both koji-builder and koji-hub: includes name and pvc size
* manage shared postgresql instance for testing
* manage shared config for mbs-frontend and mbs-backend
* change koji-hub and koji-builder to not  create root CA and PVC resources - they should fail in case those are missing
* Update docs with regards to this CRD usage

## Definitioin of Done

Mbox CR documentation and managing shared resources between koji-hub and koji-builder.

### PR DESCRIPTION
**Pull request by odra**
Fixes #33 

## Proposed Changes

* Adds mbs shared config in mbox cr
* Adds mbox crd docs

## Verification Steps

* Run tests
* Build docs

### CODE REVIEW

File: docs/source/user-guide/mbox.rst [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+Description
+===========
+
+Mbox is a configuration resource that can be optionally used to defined shared configuration across other components.
+
+Dependencies
+============
+
+`Kojira Custom Resource Definition (CRD) <https://raw.githubusercontent.com/fedora-infra/mbbox/master/mbox-operator/deploy/crds/apps.fedoraproject.org_mbox_crd.yaml>`_
```
Comment (Zlopez) [inline docs/source/user-guide/mbox.rst:15]:
```suggestion
`MBox Custom Resource Definition (CRD) <https://raw.githubusercontent.com/fedora-infra/mbbox/master/mbox-operator/deploy/crds/apps.fedoraproject.org_mbox_crd.yaml>`_
```

File: docs/source/user-guide/mbox.rst [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+-------------
+
+The koji pvc size to be used as shared volume across components.
+
+This value will be ignore if using an existing volume instead of creating one.
+
+root_ca_secret_name
+-------------------
+
+Root CA secret used to generated certificates across many components (koji clients, httpd, etc).
```
Comment (Zlopez) [inline docs/source/user-guide/mbox.rst:87]:
```suggestion
Root CA secret used to generate certificates across many components (koji clients, httpd, etc).
```

File: docs/source/user-guide/mbox.rst [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        fillme
+      cert: -|
+        fillme
+      key: -|
+        fillme 
+
+koji_hub_host
+-------------
+
+The koji-hub internal service address (service name) and port to be used across many koji and mbs components.
```
Comment (Zlopez) [inline docs/source/user-guide/mbox.rst:111]:
```suggestion
The koji-hub internal service address (service name) and port to be used across koji and mbs components.
```

File: docs/source/user-guide/mbox.rst [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+mq_topic_prefix
+---------------
+
+The MQ topic prefix to use when liseting/emitting messages.
```
Comment (Zlopez) [inline docs/source/user-guide/mbox.rst:116]:
```suggestion
The MQ topic prefix to use when lisening/emitting messages.
```

File: docs/source/user-guide/mbs-backend.rst [hunk 5]
```diff
@@ -199,7 +199,7 @@ config_pdc_url
 Configures the URL for the Product Definition Centre
 
 fedora_versions
---------------
+---------------
 
 The versions for the Fedora 32
```
Comment (Zlopez) [inline docs/source/user-guide/mbs-backend.rst:204]:
```suggestion
The versions of the Fedora we need to generate module template for.
```

Isn't this also something used by both MBS components?

**Comment by Zlopez** [comment]
Why adding `.gitkeep` to `docs/source/_static`?

File: docs/source/user-guide/mbox.rst [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+| mbs                  | {}                                            | dict    |
++----------------------+-----------------------------------------------+---------+
+| mbs.scm_repo_url     | git+https://src.fedoraproject.org/modules/    | string  |
++----------------------+-----------------------------------------------+---------+
+| mbs.rpm_repo_url     | git+https://src.fedoraproject.org/rpms/       | string  |
++----------------------+-----------------------------------------------+---------+
+| mbs. pkg_repo_url    | https://src.fedoraproject.org/repo/pkgs/      | string  |
++----------------------+-----------------------------------------------+---------+
+| mbs.pdc_url          | https://pdc.stg.fedoraproject.org/rest_api/v1 | string  |
++----------------------+-----------------------------------------------+---------+
```
Comment (Zlopez) [inline docs/source/user-guide/mbox.rst:45]: Fedora versions are missing in mbox documentation.

File: mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml [hunk 7]
```diff
@@ -7,4 +7,12 @@ spec:
   koji_pvc_name: mbox-koji-mnt
   koji_pvc_size: 10Gi
   root_ca_secret_name: mbox-koji-root-ca
-  koji_hub_host: koji-hub:8443
\ No newline at end of file
+  koji_hub_host: koji-hub:8443
+  mq_topic_prefix: 'org.fedoraproject.dev'
+  mbs:
```
Comment (Zlopez) [inline mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml:12]: Looking at the https://github.com/fedora-infra/mbbox/blob/master/mbox-operator/roles/mbs-frontend/templates/mbs-configmap.yaml.j2 I think we missed plenty of shared variables.

**Comment by odra** [comment]
ok, let me add the rest of the vars from that configmap resource

**Comment by odra** [comment]
I added _static because sphinx was throwing a warning it didn't exist.

**Comment by odra** [comment]
The docs build process was also failing, we might want to change the docs build ci test command.

**Comment by Zlopez** [comment]
The last run I see passed.

File: docs/source/deployment-guide.rst [hunk 8]
```diff
@@ -43,7 +43,7 @@ Prepare PostgreSQL DB
 
 In case you have PostgreSQL DB running in cluster you can skip this step and just use the existing one.
 
-To deploy PostgreSQL DB you can use the one `prepared by us <https://github.com/fedora-infra/mbbox/tree/master/components/psql>`_. You can change anything in those files, especially secret file. To deploy it run the following.
+To deploy PostgreSQL DB you can use the one `prepared by us <https://github.com/fedora-infra/mbbox/tree/master/components/psql>`__. You can change anything in those files, especially secret file. To deploy it run the following.
```
Comment (Zlopez) [inline docs/source/deployment-guide.rst:46]: Why this change?

File: docs/source/deployment-guide.rst [hunk 9]
```diff
@@ -54,7 +54,7 @@ Prepare RabbitMQ server
 
 In case you have a running RabbitMQ server in your cluster, you can skip this step and just use the existing one.
 
-To deploy RabbitMQ you can use the one `prepared by us <https://github.com/fedora-infra/mbbox/tree/master/components/rabbitmq>`_. You can change anything in those files, especially secret file. Refer to the `README.md` file for instructions about certificates. To deploy it run the following.
+To deploy RabbitMQ you can use the one `prepared by us <https://github.com/fedora-infra/mbbox/tree/master/components/rabbitmq>`__. You can change anything in those files, especially secret file. Refer to the `README.md` file for instructions about certificates. To deploy it run the following.
```
Comment (Zlopez) [inline docs/source/deployment-guide.rst:57]: Same as above.

File: docs/source/user-guide/mbox.rst [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+  metadata:
+    name: example
+  spec:
+    psql_secret_name: postgres
+    koji_pvc_name: mbox-koji-mnt
+    koji_pvc_size: 10Gi
+    root_ca_secret_name: mbox-koji-root-ca
+    koji_hub_host: koji-hub:8443
+    mq_topic_prefix: 'org.fedoraproject.dev'
+    mbs:
```
Comment (Zlopez) [inline docs/source/user-guide/mbox.rst:195]: This is not corresponding to the CR now.

File: docs/source/user-guide/mbox.rst [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
++------------------------------+-----------------------------------------------+---------+
+| koji_pvc_size                | 10Gi                                          | string  |
++------------------------------+-----------------------------------------------+---------+
+| root_ca_secret_name          | mbox-koji-root-ca                             | string  |
++------------------------------+-----------------------------------------------+---------+
+| koji_hub_host                | koji-hub:8443                                 | string  |
++------------------------------+-----------------------------------------------+---------+
+| mq_topic_prefix              | org.fedoraproject.dev                         | string  |
++------------------------------+-----------------------------------------------+---------+
+| mbs                          | {}                                            | dict    |
```
Comment (Zlopez) [inline docs/source/user-guide/mbox.rst:36]: I'm still missing a few mentioned in configmap for [MBS frontend](https://github.com/fedora-infra/mbbox/blob/master/mbox-operator/roles/mbs-frontend/templates/mbs-configmap.yaml.j2).

Here is the list, so we have something we can work with:
* mbs_fedora_versions
* mbs_hub_host
* mbs_messaging_system
* mbs_topic_prefix
* mbs_scm_url
* mbs_rpms_default_repository
* mbs_rpms_default_cache
* mbs_modules_default_repository
* mbs_psql_user
* mbs_psql_pass
* mbs_psql_hostname
* mbs_psql_db_name
* mbs_pdc_url
* mbs_oidc_required_scope

I hope I didn't missed anything.

Comment (odra) [reply docs/source/user-guide/mbox.rst:36]: those vars are alread there as the mbs dict property (mbs.something).

psl uses the global one

Comment (Zlopez) [reply docs/source/user-guide/mbox.rst:36]: Yeah, but I don't think all of them are in mbox CR. For example I don't see:
* hub_host
* topic_prefix
* rpms_default_repository
* rpms_default_cache
* modules_default_repository
* oidc_required_scope

File: mbox-operator/roles/mbs-backend/tasks/main.yml [hunk 12]
```diff
@@ -12,8 +12,42 @@
...
+        mbs_backend_config_rpms_def_cache: "{{ k8s_mboxes.resources[0].spec.mbs.pkg_repo_url }}"
+      when: k8s_mboxes.resources[0].spec.mbs.pkg_repo_url is defined
+    - set_fact:
+        mbs_backend_config_pdc_url: "{{ k8s_mboxes.resources[0].spec.mbs.pdc_url }}"
+      when: k8s_mboxes.resources[0].spec.mbs.pdc_url is defined
+    - set_fact:
+        mbs_backend_config_fedora_versions: "{{ k8s_mboxes.resources[0].spec.mbs.fedora_versions }}"
+      when: k8s_mboxes.resources[0].spec.mbs.fedora_versions is defined
+    - set_fact:
+        mbs_backend_hub: "{{ k8s_mboxes.resources[0].spec.mbs.backend_hub }}"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-backend/tasks/main.yml:34]: I don't see this in MBox CR.

File: mbox-operator/roles/mbs-backend/tasks/main.yml [hunk 13]
```diff
@@ -12,8 +12,42 @@
...
+      when: k8s_mboxes.resources[0].spec.mbs.backend_hub is defined
+    - set_fact:
+        mbs_backend_config_system: "{{ k8s_mboxes.resources[0].spec.mbs.mbs_backend_config_system }}"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-backend/tasks/main.yml:37]: I don't see this in MBox CR.

File: mbox-operator/roles/mbs-backend/tasks/main.yml [hunk 14]
```diff
@@ -12,8 +12,42 @@
...
+      when: k8s_mboxes.resources[0].spec.mbs.mbs_backend_config_system is defined
+    - set_fact:
+        mbs_backend_config_messaging: "{{ k8s_mboxes.resources[0].spec.backend_config_messaging }}"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-backend/tasks/main.yml:40]:
```suggestion
        mbs_backend_config_messaging: "{{ k8s_mboxes.resources[0].spec.mbs.backend_config_messaging }}"
```

File: mbox-operator/roles/mbs-frontend/tasks/main.yml [hunk 15]
```diff
@@ -12,8 +12,30 @@
...
+        mbs_pdc_url: "{{ k8s_mboxes.resources[0].spec.mbs.pdc_url }}"
+      when: k8s_mboxes.resources[0].spec.mbs.pdc_url is defined
+    - set_fact:
+        mbs_topic_prefix: "{{ k8s_mboxes.resources[0].spec.mbs.mq_topic_prefix }}"
+      when: k8s_mboxes.resources[0].spec.mbs.mq_topic_prefix
+    - set_fact:
+        mbs_fedora_versions: "{{ k8s_mboxes.resources[0].spec.mbs.fedora_versions }}"
+      when: k8s_mboxes.resources[0].spec.mbs.fedora_versions
+    - set_fact:
+        mbs_messaging_system: "{{ k8s_mboxes.resources[0].spec.backend_config_messaging }}"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-frontend/tasks/main.yml:37]:
```suggestion
        mbs_messaging_system: "{{ k8s_mboxes.resources[0].spec.mbs.backend_config_messaging }}"
```

File: mbox-operator/roles/mbs-frontend/tasks/main.yml [hunk 16]
```diff
@@ -12,8 +12,30 @@
...
+        mbs_rpms_default_repository: "{{ k8s_mboxes.resources[0].spec.mbs.rpm_repo_url }}"
+      when: k8s_mboxes.resources[0].spec.mbs.rpm_repo_url is defined
+    - set_fact:
+        mbs_rpms_default_cache: "{{ k8s_mboxes.resources[0].spec.mbs.pkg_repo_url }}"
+      when: k8s_mboxes.resources[0].spec.mbs.pkg_repo_url is defined
+    - set_fact:
+        mbs_pdc_url: "{{ k8s_mboxes.resources[0].spec.mbs.pdc_url }}"
+      when: k8s_mboxes.resources[0].spec.mbs.pdc_url is defined
+    - set_fact:
+        mbs_topic_prefix: "{{ k8s_mboxes.resources[0].spec.mbs.mq_topic_prefix }}"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-frontend/tasks/main.yml:31]: I don't see this in MBox CR.

File: mbox-operator/roles/mbs-frontend/tasks/main.yml [hunk 17]
```diff
@@ -12,8 +12,30 @@
       when: k8s_mboxes.resources|length == 0
   when: mbs_frontend_mbox|length > 0
 
-- set_fact:
-    postgres_secret: "{{ k8s_mboxes.resources[0].spec.psql_secret_name }}"
+- block:
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-frontend/tasks/main.yml:15]: These are missing here:
* hub_host
* rpms_default_repository
* rpms_default_cache
* modules_default_repository
* oidc_required_scope

Comment (odra) [reply mbox-operator/roles/mbs-frontend/tasks/main.yml:15]: I used the frontend vars name in here (which some differ from the ones used in backend role)

Comment (Zlopez) [reply mbox-operator/roles/mbs-frontend/tasks/main.yml:15]: This is the frontend role and I looked at the [MBS frontend configmap](https://github.com/fedora-infra/mbbox/blob/master/mbox-operator/roles/mbs-frontend/templates/mbs-configmap.yaml.j2)

Comment (Zlopez) [reply mbox-operator/roles/mbs-frontend/tasks/main.yml:15]: BTW this configmap should be in both frontend and backend.

File: docs/source/user-guide/mbox.rst [hunk 18]
```diff
@@ -0,0 +1,231 @@
...
++------------------------------+---------------------------------------------------+---------+
+| mq_topic_prefix              | org.fedoraproject.dev                             | string  |
++------------------------------+---------------------------------------------------+---------+
+| mbs                          | {}                                                | dict    |
++------------------------------+---------------------------------------------------+---------+
+| mbs.scm_repo_url             | git+https://src.fedoraproject.org/modules/        | string  |
++------------------------------+---------------------------------------------------+---------+
+| mbs.rpm_repo_url             | git+https://src.fedoraproject.org/rpms/           | string  |
++------------------------------+---------------------------------------------------+---------+
+| mbs. pkg_repo_url            | https://src.fedoraproject.org/repo/pkgs/          | string  |
```
Comment (Zlopez) [inline docs/source/user-guide/mbox.rst:42]:
```suggestion
| mbs.pkg_repo_url             | https://src.fedoraproject.org/repo/pkgs/          | string  |
```

File: mbox-operator/roles/mbs-backend/tasks/main.yml [hunk 19]
```diff
@@ -12,8 +12,42 @@
...
+      when: k8s_mboxes.resources[0].spec.mbs.backend_config_messaging is defined
+    - set_fact:
+        mbs_backend_hub_username: "{{ k8s_mboxes.resources[0].spec.backend_hub_username }}"
```
[the remark this answers is not in the document]
Comment (Zlopez) [reply mbox-operator/roles/mbs-backend/tasks/main.yml:37]: This is still missing in MBox CR.

File: mbox-operator/roles/mbs-backend/tasks/main.yml [hunk 20]
```diff
@@ -12,8 +12,39 @@
...
+        mbs_backend_config_messaging: "{{ k8s_mboxes.resources[0].spec.mbs.backend_config_messaging }}"
+      when: k8s_mboxes.resources[0].spec.mbs.backend_config_messaging is defined
+    - set_fact:
+        mbs_backend_hub_username: "{{ k8s_mboxes.resources[0].spec.backend_hub_username }}"
+      when: k8s_mboxes.resources[0].spec.backend_hub_username is defined
+    - set_fact:
+        mbs_backend_hub: "{{ k8s_mboxes.resources[0].spec.koji_hub_host }}"
+      when: k8s_mboxes.resources[0].spec.koji_hub_host is defined
+    - set_fact:
+        mbs_backend_config_topic_prefix: "{{ k8s_mboxes.resources[0].spec.mq_topic_prefix }}"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-backend/tasks/main.yml:43]: This should be in MBS section of MBox CR.

And this is not the MQ topic, because MBS doesn't use Fedora messaging, this is fedmsg topic prefix.

File: mbox-operator/roles/mbs-frontend/tasks/main.yml [hunk 21]
```diff
@@ -12,8 +12,36 @@
...
+        mbs_rpms_default_repository: "{{ k8s_mboxes.resources[0].spec.mbs.rpm_repo_url }}"
+      when: k8s_mboxes.resources[0].spec.mbs.rpm_repo_url is defined
+    - set_fact:
+        mbs_rpms_default_cache: "{{ k8s_mboxes.resources[0].spec.mbs.pkg_repo_url }}"
+      when: k8s_mboxes.resources[0].spec.mbs.pkg_repo_url is defined
+    - set_fact:
+        mbs_pdc_url: "{{ k8s_mboxes.resources[0].spec.mbs.pdc_url }}"
+      when: k8s_mboxes.resources[0].spec.mbs.pdc_url is defined
+    - set_fact:
+        mbs_topic_prefix: "{{ k8s_mboxes.resources[0].spec.mq_topic_prefix }}"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-frontend/tasks/main.yml:31]: This should be in MBS section of MBox CR.

And this is not the MQ topic, because MBS doesn't use Fedora messaging, this is fedmsg topic prefix.

File: mbox-operator/roles/mbs-frontend/tasks/main.yml [hunk 22]
```diff
@@ -12,8 +12,36 @@
...
+      when: k8s_mboxes.resources[0].spec.mbs.mq_topic_prefix
+    - set_fact:
+        mbs_fedora_versions: "{{ k8s_mboxes.resources[0].spec.mbs.fedora_versions }}"
+      when: k8s_mboxes.resources[0].spec.mbs.fedora_versions
+    - set_fact:
+        mbs_frontend_koji_hub_host: "{{ k8s_mboxes.resources[0].spec.koji_hub_host }}"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-frontend/tasks/main.yml:37]: This is named `mbs_hub_host` in configmap template.

Comment (odra) [reply mbox-operator/roles/mbs-frontend/tasks/main.yml:37]: so the vars defined in defaults are not being used?

Comment (odra) [reply mbox-operator/roles/mbs-frontend/tasks/main.yml:37]: Also I think it would make more sense if mbox role created this configmap, do yo think it makes sense?

Comment (Zlopez) [reply mbox-operator/roles/mbs-frontend/tasks/main.yml:37]: It definitely does.

Comment (Zlopez) [reply mbox-operator/roles/mbs-frontend/tasks/main.yml:37]: `mbs_frontend_koji_hub_host` is used for frontend specific configmap, not the shared one.

File: mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml [hunk 23]
```diff
@@ -7,4 +7,18 @@ spec:
...
+      - '32'
+    scm_repo_url: 'git+https://src.fedoraproject.org/modules/'
+    rpm_repo_url: 'git+https://src.fedoraproject.org/rpms/'
+    pkg_repo_url: 'https://src.fedoraproject.org/repo/pkgs/'
+    pdc_url: 'https://pdc.stg.fedoraproject.org/rest_api/v1'
+    oidc_required_scope: 'https://mbs.fedoraproject.org/oidc/submit-build'
+    config_system: koji
+    backend_config_messaging: fedmsg
+    hub_username: mbs
+    mq_topic_prefix: 'org.fedoraproject.dev'
```
Comment (Zlopez) [inline mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml:24]: I don't understand why there is one `mq_topic_prefix` in `spec` section and other in `mbs`.

File: mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml [hunk 24]
```diff
@@ -7,4 +7,18 @@ spec:
...
+    fedora_versions:
+      - '32'
+    scm_repo_url: 'git+https://src.fedoraproject.org/modules/'
+    rpm_repo_url: 'git+https://src.fedoraproject.org/rpms/'
+    pkg_repo_url: 'https://src.fedoraproject.org/repo/pkgs/'
+    pdc_url: 'https://pdc.stg.fedoraproject.org/rest_api/v1'
+    oidc_required_scope: 'https://mbs.fedoraproject.org/oidc/submit-build'
+    config_system: koji
+    backend_config_messaging: fedmsg
+    hub_username: mbs
```
Comment (Zlopez) [inline mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml:23]: This variable is not used.

File: mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml [hunk 25]
```diff
@@ -7,4 +7,18 @@ spec:
...
+  mbs:
+    configmap: mbs-configmap
+    fedora_versions:
+      - '32'
+    scm_repo_url: 'git+https://src.fedoraproject.org/modules/'
+    rpm_repo_url: 'git+https://src.fedoraproject.org/rpms/'
+    pkg_repo_url: 'https://src.fedoraproject.org/repo/pkgs/'
+    pdc_url: 'https://pdc.stg.fedoraproject.org/rest_api/v1'
+    oidc_required_scope: 'https://mbs.fedoraproject.org/oidc/submit-build'
+    config_system: koji
```
Comment (Zlopez) [inline mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml:21]: This variable is not used.

File: mbox-operator/roles/mbox/tasks/main.yml [hunk 26]
```diff
@@ -49,3 +49,44 @@
...
+        namespace: "{{ meta.namespace }}"
+        name: "{{ mbox_mbs.configmap }}"
+      register: k8s_mbs_cm
+    - template:
+        src: mbs.configmap.yaml.j2
+        dest: "{{ mbox_dir.path }}/mbs.configmap.yaml"
+      vars:
+        mbs_configmap: "{{ mbox_mbs.configmap }}"
+        mbs_fedora_versions: "{{ mbox_mbs.fedora_versions }}"
+        mbs_hub_host: "{{ mbox_koji_hub_host }}"
```
Comment (Zlopez) [inline mbox-operator/roles/mbox/tasks/main.yml:67]: This needs to be in format `https://koji-hub:8443`, but it's currently only `koji-hub:8443`. There are two options now:

1. We can assume that only HTTPS is supported and add it to configmap template
2. We can add another mbs value, that will allow user to also specify `http://` or `https://`. This is the solution I currently have in MBS Frontend.

File: mbox-operator/roles/mbs-backend/tasks/main.yml [hunk 27]
```diff
@@ -30,7 +30,6 @@
     - set_fact:
         psql_secret: "{{ k8s_psql_secrets.resources[0] }}"
 
-
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-backend/tasks/main.yml:33]: We should split the configmap for MBS backend, so it could benefit from the shared resource.

File: mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml (hunk 23 above)
Comment (odra) [reply mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml:24]: you said mbs needed to use a field of its own because it used fedmsg

File: mbox-operator/roles/mbox/tasks/main.yml (hunk 26 above)
Comment (odra) [reply mbox-operator/roles/mbox/tasks/main.yml:67]: I will hardcode https since we had http issues in the past

File: mbox-operator/roles/mbs-backend/tasks/main.yml (hunk 27 above)
Comment (odra) [reply mbox-operator/roles/mbs-backend/tasks/main.yml:33]: I am confused now :)

File: mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml (hunk 23 above)
Comment (Zlopez) [reply mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml:22]: Yeah, but what component is using the `mq_topic_prefix` in spec section?

File: mbox-operator/roles/mbs-backend/tasks/main.yml (hunk 27 above)
Comment (Zlopez) [reply mbox-operator/roles/mbs-backend/tasks/main.yml:33]: When @AdamSaleh created a configmap for MBS backend, he joined together MBS backend [specific configmap](https://github.com/fedora-infra/mbbox/blob/0ce9abaf857a73227719a16482fc88e77061b492/components/mbs_backend/configmap.yml) and [shared configmap](https://github.com/fedora-infra/mbbox/blob/0ce9abaf857a73227719a16482fc88e77061b492/components/mbs_shared/configmap.yml).

These two should be separated again, so the shared part could be used from mbox component.

File: mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml (hunk 23 above)
Comment (odra) [reply mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml:22]: used bu kojihub

Comment (Zlopez) [reply mbox-operator/deploy/crds/apps.fedoraproject.org_v1alpha1_mbox_cr.yaml:22]: Ok, in this case we should distinct them at least, what about using `koji_topic_prefix` and `mbs.topic_prefix` so they are easily recognized.

File: mbox-operator/roles/mbs-backend/tasks/main.yml (hunk 27 above)
Comment (odra) [reply mbox-operator/roles/mbs-backend/tasks/main.yml:33]: ok but I think it should be addressed in another issue rather than in this one, looks like mbs needs some kind of design review/refactor.

Comment (Zlopez) [reply mbox-operator/roles/mbs-backend/tasks/main.yml:33]: Yeah, no problem, we could do it in separate issue. It shouldn't be much work anyway.

Comment (Zlopez) [reply mbox-operator/roles/mbs-backend/tasks/main.yml:33]: https://github.com/fedora-infra/mbbox/issues/117 Issue created

File: docs/source/user-guide/mbox.rst [hunk 28]
```diff
@@ -0,0 +1,232 @@
...
++------------------------------+---------------------------------------------------+----------+
+| mq_topic_prefix              | org.fedoraproject.dev                             |  string  |
++------------------------------+---------------------------------------------------+----------+
+| mbs                          | {}                                                |  dict    |
++------------------------------+---------------------------------------------------+----------+
+| mbs.scm_repo_url             | git+https://src.fedoraproject.org/modules/        |  string  |
++------------------------------+---------------------------------------------------+----------+
+| mbs.rpm_repo_url             | git+https://src.fedoraproject.org/rpms/           |  string  |
++------------------------------+---------------------------------------------------+----------+
+| mbs. pkg_repo_url            | https://src.fedoraproject.org/repo/pkgs/          |  string  |
```
Comment (Zlopez) [inline docs/source/user-guide/mbox.rst:42]:
```suggestion
| mbs.pkg_repo_url             | https://src.fedoraproject.org/repo/pkgs/          |  string  |
```

File: docs/source/user-guide/mbox.rst [hunk 29]
```diff
@@ -0,0 +1,232 @@
...
++------------------------------+---------------------------------------------------+----------+
+| mbs.pdc_url                  | https://pdc.stg.fedoraproject.org/rest_api/v1     |  string  |
++------------------------------+---------------------------------------------------+----------+
+| mbs.backend_config_messaging | fedmsg                                            |  string  |
++------------------------------+---------------------------------------------------+----------+
+| mbs.fedora_versions          | ['32']                                            | [string] |
++------------------------------+---------------------------------------------------+----------+
+| mbs.mq_topic_prefix          |  org.fedoraproject.dev'                           |  string  |
```
Comment (Zlopez) [inline docs/source/user-guide/mbox.rst:50]:
```suggestion
| mbs.mq_topic_prefix          |  'org.fedoraproject.dev'                          |  string  |
```

**Comment by Zlopez** [comment]
I'm still in favour to rename `mq_topic_prefix` to `topic_prefix` inside mbs section.

**Comment by odra** [comment]
done
